File: frontend/src/types/acquisition.ts
export interface AcquisitionRunSummary {
  acquisition_id: string;
  manifest_id: string;
  status: AcquisitionStatus;
  started_at: string;
  total_sources: number;
  completed: number;
  failed: number;
}

export interface AcquisitionRunDetail extends AcquisitionRunSummary {
  elapsed_seconds: number;
  pending: number;
  retrying: number;
}

export interface AcquisitionSourceStatus {
  source_id: string;
  name: string;
  regulatory_body: string;
  access_method: string;
  status: SourceAcqStatus;
  duration_ms: number | null;
  staged_document_id: string | null;
  error: string | null;
  retry_count: number;
}

export interface StartAcquisitionResponse {
  acquisition_id: string;
  manifest_id: string;
  status: string;
  total_sources: number;
  stream_url: string;
}

export interface AcqSSEEvent {
  source_id?: string;
  name?: string;
  method?: string;
  staged_id?: string;
  duration_ms?: number;
  byte_size?: number;
  error?: string;
  retry_count?: number;
  completed?: number;
  failed?: number;
  pending?: number;
  retrying?: number;
  total?: number;
  message?: string;
}

export type AcquisitionStatus = "pending" | "running" | "complete" | "failed" | "cancelled";
export type SourceAcqStatus = "pending" | "running" | "complete" | "failed" | "retrying" | "skipped";

File: backend/app/schemas/acquisition.py
from datetime import datetime

from pydantic import BaseModel


class StartAcquisitionRequest(BaseModel):
    manifest_id: str


class StartAcquisitionResponse(BaseModel):
    acquisition_id: str
    manifest_id: str
    status: str
    total_sources: int
    stream_url: str


class AcquisitionRunDetail(BaseModel):
    acquisition_id: str
    manifest_id: str
    status: str
    started_at: datetime
    elapsed_seconds: float
    total_sources: int
    completed: int
    failed: int
    pending: int
    retrying: int


class AcquisitionRunSummary(BaseModel):
    acquisition_id: str
    manifest_id: str
    status: str
    started_at: datetime
    total_sources: int
    completed: int
    failed: int


class AcquisitionSourceStatus(BaseModel):
    source_id: str
    name: str
    regulatory_body: str
    access_method: str
    status: str
    duration_ms: int | None = None
    staged_document_id: str | None = None
    error: str | None = None
    retry_count: int = 0


class RetryResponse(BaseModel):
    source_id: str
    status: str
    retry_count: int
    message: str


class AcquisitionListResponse(BaseModel):
    acquisitions: list[AcquisitionRunSummary]


class AcquisitionSourcesResponse(BaseModel):
    sources: list[AcquisitionSourceStatus]
